File: frontend/src/services/bookingService.js
import apiClient from "./apiClient";

const bookingService = {
  createBooking: (bookingData) => apiClient.post("/bookings", bookingData),

  getMyBookings: (status) =>
    apiClient.get("/bookings/my", {
      params: status ? { status } : {},
    }),

  getAdminBookings: (status) =>
    apiClient.get("/bookings/admin", {
      params: status ? { status } : {},
    }),

  getBookingById: (id) => apiClient.get(`/bookings/${id}`),

  approveBooking: (id, reason) =>
    apiClient.put(`/bookings/${id}/approve`, {
      reason,
    }),

  rejectBooking: (id, reason) =>
    apiClient.put(`/bookings/${id}/reject`, {
      reason,
    }),

  cancelBooking: (id) => apiClient.put(`/bookings/${id}/cancel`),

  deleteBooking: (id) => apiClient.delete(`/bookings/${id}`),
};

export default bookingService;

File: backend/src/main/java/com/smartcampus/hub/service/BookingService.java
package com.smartcampus.hub.service;

import com.smartcampus.hub.dto.BookingRequestDTO;
import com.smartcampus.hub.dto.BookingResponseDTO;
import com.smartcampus.hub.enums.BookingStatus;

import java.util.List;

public interface BookingService {
    BookingResponseDTO createBooking(BookingRequestDTO requestDTO);
    List<BookingResponseDTO> getMyBookings(BookingStatus status);
    List<BookingResponseDTO> getAdminBookings(BookingStatus status);
    BookingResponseDTO getBookingById(Long bookingId);
    BookingResponseDTO approveBooking(Long bookingId, String reason);
    BookingResponseDTO rejectBooking(Long bookingId, String reason);
    BookingResponseDTO cancelBooking(Long bookingId);
    void deleteBooking(Long bookingId);
}
